web/packages/teleport/src/Discover/Shared/ConnectionDiagnostic/useConnectionDiagnostic.ts

```diff
@@ -123,6 +123,12 @@ export function useConnectionDiagnostic() {
     canTestConnection,
     username,
     authType,
+    /**
+     * @deprecated Get clusterId from resource, for example (agentMeta as NodeMeta).node.clusterId
+     * Alternatively, call useTeleport and then ctx.storeUser.getClusterId.
+     *
+     * Hooks should not reexport values that are already made available by other hooks.
+     */
     clusterId: ctx.storeUser.getClusterId(),
```

Comment:
This is another good example of "reverse hook drilling". `useConnectionDiagnostic` calls `useTeleport` and then passes `clusterId` up. With `useTestConnection` inlined into `TestConnection`, `TestConnection` could instead call both `useTeleport` and `useConnectionDiagnostic` instead.

We could even have `useClusterId` to avoid having to go through `useTeleport().storeUser.getClusterId`. But in this case I found out that `clusterId` can be accessed without calling any hooks. Other connection tests depend on `clusterId` from here though, so for now I just marked it as deprecated.